runtime/libc/wasi/wasi_main.c

```diff
@@ -0,0 +1,43 @@
+#include <stdlib.h>
+
+#include "../../runtime.h"
+#include "wasi_impl.h"
+
+void* current_wasi_context;
+
+
+/* Code that actually runs the wasm code */
+IMPORT void wasmf__start(void);
+
+#define unlikely(x) __builtin_expect(!!(x), 0)
+
+/**
+ * @brief Lifecycle function that runs at end at termination of WebAssembly module. Suitable for logging and cleanup
+ */
+void runtime_on_module_exit() {
+    wasi_context_destroy(current_wasi_context);
+    return;
+}
+
+int main(int argc, char* argv[]) {
```

Comment:
If this is separate from the backing implementations (uvwasi or minimal) to prevent copying it across both implementations, that's OK. But it will not be used on a microcontroller. I'd hope that it is easy to write a new version of this and the wasi implementation so taht Maorui doesn't need to wrestle with the build system.

Reply:
That's correct! It's just used in both places to keep things DRY. I added a comment to this affect to the top of the file and added a section to the README discussing this and providing an example of such a difference in SLEdge due its different execution model. The "build system wrestling" to make this work would mainly be in including the correct runtime translation units during the final compilation step.